internal/v4/archive_test.go

```diff
@@ -406,22 +404,33 @@ func (s *ArchiveSuite) TestPostCharm(c *gc.C) {
 	})
 	c.Assert(rec.Code, gc.Equals, http.StatusOK)
 	c.Assert(rec.Header().Get(params.EntityIdHeader), gc.Equals, "cs:~charmers/precise/wordpress-1")
+
+	// Retrieving the development version returns the last development charm.
+	err = s.store.SetPerms(charm.MustParseURL("~charmers/development/wordpress"), "read", params.Everyone)
+	c.Assert(err, gc.IsNil)
+	rec = httptesting.DoRequest(c, httptesting.DoRequestParams{
+		Handler: s.srv,
+		URL:     storeURL("~charmers/development/wordpress/archive"),
+	})
+	c.Assert(rec.Code, gc.Equals, http.StatusOK)
+	c.Assert(rec.Header().Get(params.EntityIdHeader), gc.Equals, "cs:~charmers/development/precise/wordpress-2")
```

Comment:
We can't do this in v4 old versions of juju cannot necessarily parse a URL containing a channel. Any URLs we return in v4 must not include channels.